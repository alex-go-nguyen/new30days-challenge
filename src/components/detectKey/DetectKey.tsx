import React, { useEffect, useState } from 'react';
import classNames from 'classnames/bind';
import styles from './DetectKey.module.scss';
import Card from '../card/Card';

const cx = classNames.bind(styles);

type KeyboardDetail = Pick<KeyboardEvent, 'key' | 'location' | 'which' | 'code'>;

export default function DetectKey() {
    const [keyDetail, setKeyDetail] = useState<KeyboardDetail>();

    const keyDownHandler = ({ key, location, which, code }: KeyboardEvent) => {
        setKeyDetail({
            key,
            location,
            which,
            code,
        });
    };

    useEffect(() => {
        document.addEventListener('keydown', keyDownHandler);
        return () => {
            document.removeEventListener('keydown', keyDownHandler);
        };
    }, []);
    return (
        <div>
            {!keyDetail ? (
                <div className={cx('empty')}>Press Any key</div>
            ) : (
                <>
                    <div className={cx('result')}>{keyDetail.which}</div>
                    <div className={cx('details')}>
                        <Card title="Key">{keyDetail.key.toLocaleUpperCase()}</Card>
                        <Card title="Location">{String(keyDetail.location)}</Card>
                        <Card title="Which">{String(keyDetail.which)}</Card>
                        <Card title="Code">{keyDetail.code}</Card>
                    </div>
                </>
            )}
        </div>
    );
}
