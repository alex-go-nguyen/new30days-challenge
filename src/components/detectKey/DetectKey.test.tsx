import React from 'react';
import renderer, { ReactTestRenderer, act } from 'react-test-renderer';
import DetectKey from './DetectKey';

describe('DetectKey component', () => {
    let component: ReactTestRenderer;
    test('should render correctly', async () => {
        await act(async () => {
            component = renderer.create(<DetectKey />);
        });
        expect(component.toJSON()).toMatchSnapshot();
    });
});
