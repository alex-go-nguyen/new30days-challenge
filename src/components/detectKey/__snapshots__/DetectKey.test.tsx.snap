// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetectKey component should render correctly 1`] = `
<div>
  <div
    className="empty"
  >
    Press Any key
  </div>
</div>
`;
